### PR TITLE
Cleaning the code

### PR DESCRIPTION
**Pull request by sdenel**
This is a first batch of modifications aiming at keeping the code as maintainable and readable as possible.

I checked that compiling with PLUGIN_UPDATER_GPU=ON is still working.

### CODE REVIEW

File: plugin/updater_gpu/src/gpu_builder.cu [hunk 1]
```diff
@@ -327,12 +327,12 @@ void GPUBuilder::Sort(int level) {
 }
 
 void GPUBuilder::Update(const std::vector<bst_gpair> &gpair, DMatrix *p_fmat,
-                        RegTree *p_tree) {
+                        RegTree &tree) {
```
Comment (tqchen) [inline plugin/updater_gpu/src/gpu_builder.cu]: pass mutable items by pointer is standard we adopt as in Google's C style

Comment (sdenel) [inline plugin/updater_gpu/src/gpu_builder.cu]: Ok, I got it.
I will iterate tomorrow in the morning and edit this PR accordingly.

Comment (sdenel) [inline plugin/updater_gpu/src/gpu_builder.cu]: This time it should be compliant!

**Comment by sdenel** [comment]
Ping! My next suggestions will be far more structural, and you might refuse them so I would prefer to create a new PR.

**Comment by tqchen** [comment]
Please make sure the travis tests pass to make sure they pass. You might want to rebase against master 

https://travis-ci.org/dmlc/xgboost/builds/178451117 

Note some of the change(class member assignment in definition) requires g++-4.8 compatibility, currently the default minimum compiler is g++-4.6

**Comment by sdenel** [comment]
This time, it should work. Is there any documented way to run travis tests locally? I wonder if we should not create a straightforward procedure to test the code. Like running docker tests from a custom docker image, that we can run locally and using an old compiler like g++-4.6.

**Comment by tqchen** [comment]
Please do a rebase, there is still errors in lint and compilation error(which need dmlc-core to get updated). You can test lint locally via make lint

**Comment by sdenel** [comment]
Thank you for the tip! This PR ended in a complete mess :)
I am closing it, and will create two different ones, one for the typos and one for OpenMP.

**Pull request closed without merging**